### DISCUSSION TITLE
Help overriding some which-key opts after update

### DISCUSSION THREAD

**Discussion by dotfrag** [LazyVim/LazyVim#4014] [Q&A] [answered]
Hello, I have [this spec](https://github.com/dotfrag/nvim/blob/main/lua/plugins/which-key.lua) which 1. adds the `regex` group and 2. overrides the default `<leader>gh` and `<leader>q` mappings.

This is not transferrable after the update, and this is as far as I got:

```lua
  {
    "folke/which-key.nvim",
    opts = {
      spec = {
        { "<leader>q", "Quit" },
        { "<leader>r", group = "regex" },
        { "<leader>h", group = "hunks" },
      },
    },
    -- config = function(_, opts)
    --   opts.defaults["<leader>gh"] = nil
    -- end,
  },
```

Regex and hunks groups work, but I do not know 1. how to remove `<leader>gh` and 2. `<leader>q` still shows `quit/session`.

Any help is appreciated.

**Comment by folke** [comment, marked as answer]
if you do `:checkhealth which-key` it will actually show you what it should be.

You need to add `desc = "Quit"`

**Reply by dotfrag** [reply]
Thanks, looks to be working now. From my understanding, I no longer to unset the current hunks group?

i.e. (old config)

```lua
opts.defaults["<leader>gh"] = nil
```

**Reply by folke** [reply]
That `defaults` option was a custom LazyVim thing.
which-key now has `opts.spec` for adding mappings.

**Reply by dotfrag** [reply]
Yes, but LazyVim has `<leader>gh` in opts.spec of which-key. Does that need to be unset in our config?

PS. For `+hunks` we can use one of the icons from the keywords "diff" or "compare" from https://www.nerdfonts.com/cheat-sheet.

**Reply by folke** [reply]
If you don't want to show `<leader>gh`, you can add

```lua
 { "<leader>gh", hidden = true },
```

**Comment by dpetka2001** [comment]
I'm going to hijack this thread, so that I don't open a new one. I'm trying the following in `opts.spec`
```lua
        { "<leader>C", group = "Ccc" },
        { "<leader>CC", "<cmd>echo 'test'<cr>", desc = "test" },
```
But when I open `which-key` menu by pressing `<leader>`, it only shows `CC -> test` as a single mapping and doesn't show the group name `C -> Ccc`. Any idea what I'm doing wrong?

**Reply by folke** [reply]
That's the `opts.expand` setting. It defaults to 1, meaning that if a group only contains one mapping, it will be expanded in the parent group.

Maybe I should set the default to 0 though...

**Reply by dpetka2001** [reply]
@folke It seems to me that when using a capital letter it doesn't create a group, but instead expects it to be a command? If i use `<leader>c` instead it correctly overrides the `code` group. Is this intended? In previous version you could create a group even if it started with a capital letter.

**Reply by dpetka2001** [reply]
Oh, I just saw your comment about `opts.expand`, so maybe my previous comment doesn't stand.

**Reply by dpetka2001** [reply]
You're correct. If I add another dummy mapping in the group it correctly shows the top-level group name.

**Reply by dpetka2001** [reply]
Either way, regardless of what you intend to do (change the default `opts.expand` or not, no idea if there might be other conflicts so it's up to you to decide what is best), now that I know the reason I don't have any problem just creating a dummy mapping. It's just one either way, so no biggie.

**Reply by dpetka2001** [reply]
Ok, I should have read more carefully. You meant `opts.expand` setting in `which-key` and I got confused with `opts_extend`, that something had to change in `lazy.nvim`. I just changed the setting in `which-key` for myself and all is good. 

Thanks a lot and will try to be more careful about what you say.
